framework/include/base/MooseError.h

```diff
@@ -104,11 +104,15 @@
 template <typename... Args>
 [[noreturn]] void mooseError(Args &&... args);
 
+class MooseVariable;
+
 namespace moose
 {
 namespace internal
 {
 
+std::string incompatVarMsg(MooseVariable & var1, MooseVariable & var2);
```

Comment:
Need a really good documentation string for this method. It's unclear what this method does from the name. In fact, I don't think it really does what the name implies (prints an incompatible variable message).